benchmarking/commons/hpo_main_simulator.py

```diff
@@ -112,85 +149,54 @@ def subsample(
-        benchmarks. If ``--benchmark`` is not given, then ``benchmark_names`` are
-        keys of this dictionary.
-        Can be nested (only for internal use).
-    :param extra_args: List of dictionaries, containing additional arguments
-        to be passed. Must contain ``name`` for argument name (without leading
-        ``"--"``), and other kwargs to ``parser.add_argument``. Optional
-    :return: ``(args, method_names, benchmark_names, seeds)``, where ``args`` is
-        result of ``parser.parse_args()``, ``method_names`` see ``methods``,
-        'benchmark_names`` see ``benchmark_definitions``, and ``seeds`` are list of
-        seeds specified by ``--num_seeds`` and ``--start_seed``
+    :param configuration: ConfigDict with parameters of the benchmark
+        Must contain all parameters from LOCAL_LOCAL_SIMULATED_BENCHMARK_REQUIRED_PARAMETERS
+    :param methods: Dictionary with method constructors.
+    :param benchmark_definitions: Definitions of benchmarks; one is selected from
+        command line arguments
+    :param post_processing: Called after tuning has finished, passing the tuner
+        as argument. Can be used for postprocessing, such as output or storage
+        of extra information
+    :param map_method_args: See above, optional
+    :param tuning_job_metadata: Metadata added to the tuner, can be used to manage results
+    :param use_transfer_learning: If True, we use transfer tuning. Defaults to
+        False
     """
-    if extra_args is None:
-        extra_args = []
-    else:
-        extra_args = extra_args.copy()
-    nested_dict = is_dict_of_dict(benchmark_definitions)
-    extra_args.extend(
-        [
-            dict(
-                name="benchmark",
-                type=str,
-                help="Benchmark to run from benchmark_definitions",
-            ),
-            dict(
-                name="verbose",
-                type=int,
-                default=0,
-                help="Verbose log output?",
-            ),
-            dict(
-                name="support_checkpointing",
-                type=int,
-                default=1,
-                help="If 0, trials are started from scratch when resumed",
-            ),
-            dict(
-                name="fcnet_ordinal",
-                type=str,
-                choices=("none", "equal", "nn", "nn-log"),
-                default="nn-log",
-                help="Ordinal encoding for fcnet categorical HPs with numeric values. Use 'none' for categorical encoding",
-            ),
-            dict(
-                name="restrict_configurations",
-                type=int,
-                default=0,
-                help="If 1, scheduler only suggests configs contained in tabulated benchmark",
-            ),
-        ]
+    configuration.check_if_all_paremeters_present(
+        LOCAL_LOCAL_SIMULATED_BENCHMARK_REQUIRED_PARAMETERS
     )
-    if nested_dict:
-        extra_args.append(
```

Comment:
You are missing this `benchmark_key` argument, this is a CL argument. It is important to get the nested dictionary variant for benchmark definitions done.

Normally, all benchmarks in this dictionary are run sequentially on the same instance. But in some cases, it makes sense to parallelize over benchmarks as well, if there are many of them.

Reply:
Good catch, will add to the extra_args.